### PR TITLE
fix: filter create/upsert in nested ops for relations that don't support create

### PR DESCRIPTION
**Pull request by Weakky**
## Overview

When a model has a relation to another model that cannot support create/upsert operations, filter out create/upsert nested operations for that model

Follow up to #1561

### CODE REVIEW

File: query-engine/core/src/schema_builder/output_types/mutation_type.rs [hunk 1]
```diff
@@ -54,24 +54,36 @@ fn create_nested_inputs(ctx: &mut BuilderContext) {
     while !(nested_create_inputs_queue.is_empty() && nested_update_inputs_queue.is_empty()) {
         // Create inputs.
         for (input_object, rf) in nested_create_inputs_queue.drain(..) {
-            let mut fields = vec![input_fields::nested_create_one_input_field(ctx, &rf)];
+            let mut fields = vec![];
 
-            append_opt(&mut fields, input_fields::nested_connect_input_field(ctx, &rf));
-            append_opt(&mut fields, input_fields::nested_connect_or_create_field(ctx, &rf));
+            if rf.related_model().supports_create_operation == true {
```
Comment (dpetrick) [inline query-engine/core/src/schema_builder/output_types/mutation_type.rs:59]: I sense a pattern with your bools 😄 This one I do more firmly insist on keeping it as `if rf.related_model().supports_create_operation`.

Comment (dpetrick) [reply query-engine/core/src/schema_builder/output_types/mutation_type.rs:59]: (same with any other follow ups)

**Review by dpetrick — APPROVED** [review]

File: query-engine/core/src/schema_builder/output_types/mutation_type.rs (hunk 1 above)
Comment (do4gr) [reply query-engine/core/src/schema_builder/output_types/mutation_type.rs]: At least its not `===` ;-)

Comment (tomhoule) [reply query-engine/core/src/schema_builder/output_types/mutation_type.rs]: `matches!(rf.related_model().supports_create_operation, true)`

**Pull request merged**